src/compiler/checker.ts

```diff
@@ -11235,7 +11235,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             return baseTypeVariable ? getIntersectionType([type, baseTypeVariable]) : type;
         }
         else {
-            return strictNullChecks && symbol.flags & SymbolFlags.Optional ? getOptionalType(type) : type;
+            return strictNullChecks && symbol.flags & SymbolFlags.Optional ? getOptionalType(type, /*isProperty*/ true) : type;

```

Comment:
I'm honestly really suspicious of every other call to this function now; I went and made `isProperty` required and about _half_ of the callers are actually working with things that are explicitly property types.

No tests change when I make everyone pass in this parameter, though, which makes me think that we don't have enough coverage of this particular option.

Reply:
This option has a default though - did you try passing `true` to all calls of this function?

Reply:
No, I mean I changed it from `isProperty = false` to `isProperty: boolean` and then examined all of the places where it error'd.

Not that this is directly required for this PR, just something I noticed.

Reply:
The `missingType` thing is only relevant here for with `exactOptionalPropertyTypes` - and that's not the most popular compiler option. So I definitely wouldn't be surprised if the test suite wouldn't be that comprehensive in this regard.